chap-cheri-riscv.tex

```diff
@@ -1008,6 +979,109 @@ \subsubsection{Exceptions to User Mode}
 %for tag loss at suitable moments (e.g., after potentially tag-stripping
 %operations such as pointer manipulation).
 
+\subsection{Capability Exception Reporting}
+\label{subsection:riscv:cheri-exception-reporting}
+
+CHERI-RISC-V extends the definition of the Trap Value CSRs, \xtval{}, to
+report capability exception details as described in
+Figure~\ref{fig-cheri-tval} (shown for XLEN=32):
+
+\begin{figure}[!h]
+\begin{center}
+\begin{bytefield}[bitwidth=\textwidth/34]{32}
+  \bitheader[endianness=big]{0,4,5,10,31} \\
+  \bitbox{21}{\textbf{WPRI}}
+  \bitbox{6}{\texttt{cap idx}}
+  \bitbox{5}{\texttt{cause}}
+\end{bytefield}
+\caption{\xtval{} register format for Capability Exception}
+\label{fig-cheri-tval}
+\end{center}
+\end{figure}
+
+\begin{description}
+\item [cause] The \texttt{cause} field reports the capability
+  exception code from Table~\ref{tab:risc-v-capability-cause}.
+\item [cap idx] The \texttt{cap idx} field reports the index of the capability register that caused the last exception.  When
+the most significant bit is set, the 5 least significant bits are used to index
+the special purpose capability register file described in
+Table~\ref{tab:risc-v-special-capability-registers}, otherwise, they index the
+general-purpose capability register file.
+\end{description}
+
+\begin{table}
+\begin{center}
+\begin{tabular}{ll}
+\toprule
+Value & Description \\
+\midrule
+0x00 & None \\
```

Comment:
Would be nice if we could generate this table from sail, but almost certainly not worth the effort.